### DISCUSSION TITLE
2021-10-12 weekly discussion

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#727] [Maintainer discussions]
### About
Each Tuesday, we start a discussion about Mathesar's progress. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

### Discussion topics
- [New planning format and GitHub project](https://github.com/centerofci/mathesar/discussions/727#discussioncomment-1467414)
- [Welcome Sean!](https://github.com/centerofci/mathesar/discussions/727#discussioncomment-1467415)
- [Weekly check-in](https://github.com/centerofci/mathesar/discussions/727#discussioncomment-1467416)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/727#discussioncomment-1467418)
- [Team event](https://github.com/centerofci/mathesar/discussions/727#discussioncomment-1467420)

**Comment by kgodey** [comment]
## New planning format and GitHub project
We are switching to a [new GitHub project](https://github.com/orgs/centerofci/projects/1) to track Mathesar work in lieu of our previous Active, Backlog, and Future projects. This is a new GitHub feature that allows us a lot more flexibility to see different views of issues. Unfortunately, the new project is not yet publicly visible (it is only visible to Core and Community team members), but all the important information (status, milestone, etc.) should also be reflected in individual issues. The documentation on the wiki has also been updated to refer to the new projects

As part of the new project workflow, we are also switching to a new planning format. Instead of planning out each week's work in advance, Mathesar core team members will pick up tasks from the backlog as needed.

We will no longer be posting Active Work and Retrospective sections in the weekly discussion since we have a constantly updated [Active view](https://github.com/orgs/centerofci/projects/1/views/1) and a [Closed view](https://github.com/orgs/centerofci/projects/1/views/5) that shows recently closed issues. We will also not be posting an Updates section since the discussions should be a lot shorter; each update will get its own section.

**Comment by kgodey** [comment]
## Welcome Sean!

Our new frontend engineer, @seancolsen, has joined the Mathesar team as of today. Welcome @seancolsen, and please tell the community a little about yourself. :)

**Reply by seancolsen** [reply]
Hi everyone! I'm a frontend dev who loves Svelte. I live in Knoxville TN, USA. When I'm not coding, I'm busy wrangling my two 18 month-old twins and wondering when they'll be old enough to go mountain biking with me. I'm excited to be here and help build Mathesar!

**Reply by ghislaineguerin** [reply]
Welcome @seancolsen !

**Comment by kgodey** [comment]
## Weekly check-in

@mathemancer, @ghislaineguerin, @pavish, @seancolsen, you are encouraged to respond. Others are also welcome to respond.

- What went well over the last week?
- What could have gone better over the last week?
- What are you planning to work on over the next week?
- Do you anticipate any blockers? Can anyone else on the team help?
- Anything else you'd like to add?

**Reply by seancolsen** [reply]
> What are you planning to work on over the next week?

- [Move filter pane to dropdown](https://github.com/centerofci/mathesar/issues/643)
- Plus whatever others assign me

> Anything else you'd like to add?

Excited to be here!

**Reply by mathemancer** [reply]
> What went well over the last week?

I'm glad to get the URI type in, and some of the server default stuff done.  I think switching contexts more often is starting to work better.

> What could have gone better over the last week?

I think my velocity is a little slow.  I'm trying to do more work earlier in the day when I'm better able to focus to counteract this.

> What are you planning to work on over the next week?

I'm a bit blocked on type implementations, so I'm going to work on some of the grouping / filtering for a bit for a change.  I may also do some probably-conflicting work on the `TIMESTAMP` type.

> Do you anticipate any blockers? Can anyone else on the team help?

The main issue is that some of the types need to be implemented in a consistent way with each other, which makes parallelization difficult.

> Anything else you'd like to add?

Nothing at the moment.

**Reply by kgodey** [reply]
> What went well over the last week?

I'm glad to have finished the new project work. I'm also glad that @seancolsen is here!

> What could have gone better over the last week?

I haven't had any time for Mathesar code work because of meetings and lots of minor tasks. I hope this week will be better.

> What are you planning to work on over the next week?

Other than the day-to-day stuff,
- Refactoring connection management
  - #280 
  - #592  
- Refactoring columns
  - #658  
- Speccing out #661

> Do you anticipate any blockers? Can anyone else on the team help?

No notable blockers, but I may not respond to messages as quickly when I'm focusing on code.

**Reply by ghislaineguerin** [reply]
> What went well over the last week?

Glad that the prototype is at a state where I can demonstrate entire flows. I also think the student project is off to a good start.

> What could have gone better over the last week?

I wish I had more time to organize future work, define more issues for future work.

> What are you planning to work on over the next week?

Close review for views milestone, assist w student project and UI styling. There are also issues that I need to address related to implementation of existing designs.

> Do you anticipate any blockers? Can anyone else on the team help?

None

> Anything else you'd like to add?

Nothing for now

**Reply by pavish** [reply]
> What went well over the last week?

Work has been progressing steadily, milestone 5 is close to completion.

> What could have gone better over the last week?

I planned to create more issues, didn't manage to complete that. I think I should focus a bit on creating frontend issues for the next couple milestones.

> What are you planning to work on over the next week?

I am planning to
* Complete integration testing for the frontend, for all the features done so far.
* Close all open frontend PRs.

> Do you anticipate any blockers? Can anyone else on the team help?

None, at the moment.

> Anything else you'd like to add?

None, at the moment.

**Reply by kgodey** [reply]
@mathemancer Let me know if I can help unblock type implementations. Also, what is the plan with the custom `MONEY` type?

@ghislaineguerin I think it's okay to take some time to define issues for future work. Let me know if I can help you prioritize to make time for for that.

@pavish Does your plan for this week also involve making frontend issues for the next couple of milestones?

**Reply by mathemancer** [reply]
I think there's just a cascade of dependencies in the time types.  For the custom `MONEY` type, I'd like to wait till we have solid display options, then use those.  My thinking is that even if that gets corrupted somehow, it doesn't actually lose any data (i.e., the underlying data won't be modified).  I recognize that it's not ideal, since the user might have to reconfigure that, but it's the best long-term solution once we have a more solid column id.

On the column ID front, I did some thinking and we should be able to implement it using PostgreSQL event triggers to manage our own `mathesar_catalog` table in our reserved schema in the DB.  Then, the service could reflect from there whenever getting columns.  It will be a tricky task, but it should be possible.  The event trigger functionality should keep us out of trouble on conflicts and transactions and so on.

**Reply by kgodey** [reply]
@mathemancer Makes sense re: types.

Can you add your idea for column implementation to #658?

**Reply by pavish** [reply]
> @pavish Does your plan for this week also involve making frontend issues for the next couple of milestones?

@kgodey Yes, I will prioritize this over closing the open frontend PRs. Some of them need additional commits and further review, it might take a while to get them all merged.

**Comment by kgodey** [comment]
## Comms Assignee schedule

I think it would be better to move the Comms Assignee schedule from Monday-Friday every week rather than Tuesday-Monday for ease of remembering. Here's the updated schedule for the next five weeks:

| Week | Person |
|-|-|
| Oct 11 to Oct 15 | @mathemancer |
| Oct 18 to Oct 22 | @kgodey |
| Oct 25 to Oct 29 | @pavish |
| Nov 1 to Nov 5 | @seancolsen |
| Nov 8 to Nov 12 | @ghislaineguerin |

**Comment by kgodey** [comment]
## Team event

We need someone to organize the team event for the week of Oct 18-22. Any volunteers?

**Reply by seancolsen** [reply]
Other than the online escape room, what have some previous team events been?

**Reply by mathemancer** [reply]
We've watched a movie, played GeoGuesser, and done some trivia challenges.

**Reply by kgodey** [reply]
We don't necessarily need a pre-planned activity, but we are not the most naturally talkative of groups, so it's probably helpful to have some sort of topic or activity.

**Reply by ghislaineguerin** [reply]
@kgodey I will organize it.

**Reply by kgodey** [reply]
Thanks @ghislaineguerin! 😁

**Reply by pavish** [reply]
@kgodey I'll organize the next team event, after Ghislaine.

**Reply by kgodey** [reply]
Thanks @pavish
